src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -130,6 +130,45 @@ static Object* arraySpeciesCreate(ExecutionState& state, Object* originalArray,
     return Object::construct(state, C, 1, argv);
 }
 
+// http://ecma-international.org/ecma-262/10.0/#sec-flattenintoarray
+// FlattenIntoArray(target, source, sourceLen, start, depth [ , mapperFunction, thisArg ])
+static int64_t flattenIntoArray(ExecutionState& state, Value target, Value source, int64_t sourceLen, int64_t start, double depth, Value mappedValue = Value(), Value thisArg = Value())
+{
+    ASSERT(target.isObject());
+    ASSERT(source.isObject());
+    ASSERT(sourceLen >= 0);
+
+    int64_t targetIndex = start;
+    int64_t sourceIndex = 0;
+
+    while (sourceIndex < sourceLen) {
+        String* p = Value(sourceIndex).toString(state);
+        ObjectHasPropertyResult exists = source.asObject()->hasIndexedProperty(state, p);
+        if (exists) {
+            Value element = exists.value(state, ObjectPropertyName(state, p), source);
+            if (!mappedValue.isUndefinedOrNull()) {
+                Value args[] = { element, Value(sourceIndex), source };
+                ASSERT(mappedValue.isCallable() && depth == 1);
```

Comment:
From the standard
> If mapperFunction is present, then
    Assert: thisArg is present.
    Set element to ? Call(mapperFunction, thisArg , « element, sourceIndex, source »).

The ASSERT statement should check if the thisArg is valid or not (We can check it by `isEmpty` if thisArg has empty value in default). IMO check for callable value or depth is unnecessary.